dnschecker.go

```diff
@@ -0,0 +1,127 @@
+package checkup
+
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"time"
+	"github.com/miekg/dns"
+
+)
+
+// DNSChecker implements a Checker for TCP endpoints.
+type DNSChecker struct {
+	// Name is the name of the endpoint.
+	Name string `json:"endpoint_name"`
+
+	// This is the name of the DNS server you are testing.
+	DNSServer string `json:"endpoint_url"`
+
+	// This is the fqdn of the target server to query the DNS server for.
+	DNSTarget string `json:"hostname_fqdn,omitempty"`
+
+
```

Comment:
Can you please run `go fmt` on your code files? That will remove the extra whitespace and format some other things.